### DISCUSSION TITLE
Registration of new devices on Routers only possible when learning mode is active only on the specific router

### DISCUSSION THREAD

**Discussion by riesi77** [Koenkk/zigbee2mqtt#19798] [Devices]
I run zigbee2mqtt with a SONOFF Dongle E with a couple of Nous A1Z and TuYa TS011F_plug_1 plugs.
Sometimes a plug falls off the net and I have to re-register them again. The coordinator is not reachable from that position, just another plug. However, when I start the learning mode (all), the plug does not register. Only if I specifically activate the learning mode for the plug that acts as router.
Can someone explain why?

Thanks!

**Comment by riesi77** [comment]
Is someone experiencing the same behaviour? Also, I would expect that after some hours/days, the net would stabilize/optimize itself, meaning that plugs would then communicate over plugs nearby. But this is not the case. I have one plug, that has another one in sight, but still tries to communicate with the coordinator that is barely reachable.
I'm tempted to give a Dongle P a chance, but I don't want to waste the money and effort if it does not change anything.

**Comment by michapr** [comment]
Yes, I can confirm this.
Have same issue with thermostat and different routing devices (plug, repeaters).
Thermostat will not reach a new connection (any router). And re-register with "all" will not work. But if I will define any repeater near the thermostat it will work.
It's a bit strange.

Using the Z-Wave.Me Z-Station  (Zigbee/Z-Wave) based on Silicon Labs EFR32MG21P

**Comment by cornholio1234** [comment]
I've got the Z-Station too. My experiences so far:
Powered nodes try to connect to the controller directly always making it hard to work with longer distances (register all is working for me...but taking bad choices imho). 
But I can use register my powered nodes to closer routing nodes manually by enabling register only for them. This made some of my nodes going from LQI <50 to LQI nearly 100

![image](https://github.com/user-attachments/assets/d0e2f652-7c59-4b5f-8201-4d5714f9c3c8)

**Reply by michapr** [reply]
Yes, 100% agree - same for me.
Additional, battery devices will not register at all (by default), even if any powered node is near the device. I must specify a near routing device, then all is fine.
I think that this should be done by Zigbee software - select the best routing device for such cases.
